h/authorization.py

```diff
@@ -32,40 +32,62 @@
 their client credentials, but they do not confer any authorizations and no
 mechanism exists yet for users to grant authorizations to clients.
 """
-from annotator import auth
+import datetime
+
+import jwt
+from oauthlib.oauth2 import BearerToken
 from oauthlib.oauth2 import ClientCredentialsGrant
 from oauthlib.oauth2 import RequestValidator as _RequestValidator
 from pyramid.authorization import ACLAuthorizationPolicy
 from pyramid.exceptions import BadCSRFToken
 from pyramid.session import check_csrf_token
 
-from .oauth import AnnotatorToken
+from .oauth import JWTBearerGrant
 from .security import WEB_SCOPES
 
-
-try:
-    # pylint: disable=no-name-in-module
-    from hmac import compare_digest as is_equal
-except ImportError:
-    def is_equal(lhs, rhs):
-        """Returns True if the two strings are equal, False otherwise.
-
-        The comparison is based on a common implementation found in Django.
-        This version avoids a short-circuit even for unequal lengths to reveal
-        as little as possible. It takes time proportional to the length of its
-        second argument.
-        """
-        result = 0 if len(lhs) == len(rhs) else 1
-        lhs = lhs.ljust(len(rhs))
-        for x, y in zip(lhs, rhs):
-            result |= ord(x) ^ ord(y)
-        return result == 0
+DEFAULT_TTL = 3600
+LEEWAY = 240  # allowance for clock skew in verification
 
 
 def is_web_client(request, client_id):
     return client_id == request.registry.web_client.client_id
 
 
+def token_generator(request):
+    """
+    Generate a token from a request.
+
+    The request must have the ``client`` and `extra_credentials`` properties
+    added by OAuthLib, and a ``user`` (possibly ``None``).
+    """
+    client = request.client
+    now = datetime.datetime.utcnow().replace(microsecond=0)
+    ttl = datetime.timedelta(seconds=DEFAULT_TTL)
+
+    payload = {
+        'aud': request.host_url,
```

Comment:
I know what these fields mean, but only because I've worked with JWTs before. It might be worth adding comments to explain them, or at least a link to the relevant part of the JWT RFC.